src/frontend/src/lib/stores/user-profile.store.ts

```diff
@@ -0,0 +1,24 @@
+import type { UserProfile } from '$declarations/backend/backend.did';
+import { writable, type Readable } from 'svelte/store';
+
+// * `undefined` means the store is not loaded yet.
+// * `null` means there was an error.
+// * `UserProfile` is the data.
+export type UserProfileStoreData = UserProfile | undefined | null;
+
+export interface UserProfileStore extends Readable<UserProfileStoreData> {
+	set: (data: UserProfileStoreData) => void;
+	reset: () => void;
+}
+
+const initUserProfileStore = (): UserProfileStore => {
+	const { subscribe, set } = writable<UserProfileStoreData>(undefined);
+
+	return {
+		set: (data: UserProfileStoreData | null) => set(data),
+		reset: () => set(undefined),
```

Comment:
`set(null)`

Reply:
This would mean that there was an error. `reset` for me is to go back to the initial state, which is set with `writable<UserProfileStoreData>(undefined)`.

Does that make sense?

I can change it though.

Reply:
Please do. In Oisy, reset === null

Reply:
done!